DiagnosticsExtension/Controllers/ConnectionStringValidationController.cs

```diff
@@ -72,25 +75,24 @@ public async Task<HttpResponseMessage> Validate([FromBody] ConnectionStringReque
             {
                 return Request.CreateErrorResponse(HttpStatusCode.BadRequest, "Type is not specified in the request body");
             }
-
             var result = await Validate(requestBody.ConnectionString, requestBody.Type);
             return result;
         }
 
         [HttpGet]
         [Route("validateappsetting")]
-        public async Task<HttpResponseMessage> ValidateAppSetting(string appSettingName, string type)
+        public async Task<HttpResponseMessage> ValidateAppSetting(string appSettingName, string type, string entityName = null)
```

Comment:
This function is very specified for function apps, because if I want to validate connection strings stored in AppSettings for WebApp, I won't be able to use this function (otherwise it will treat a non-existed appsetting as MSI enabled). Maybe we can change the function name to ValidateAppSettingForFunctionApp or something like that?

Reply:
Can we add some comment to the function and briefly describe what is the `entityName` in the parameter

Reply:
I assume you would also want the route to change from "validateappsetting" to something more functions specific?

Reply:
yep